### PR TITLE
Removing Coverage from Nightly Builds (other than py2.7 and py3.7)

### PR DESCRIPTION
**Pull request by scbedd**
Coverage is fairly heavy on pytest execution. This is an attempt to reduce the load a bit.

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: eng/pipelines/templates/jobs/archetype-sdk-nightly.yml [hunk 1]
```diff
@@ -88,38 +72,52 @@ jobs:
...
+          PythonVersion: '3.6'
+        Linux_Python35:
+          OSName: 'Linux'
+          OSVmImage: 'ubuntu-16.04'
+          PythonVersion: '3.5'
+        Linux_PyPy3:
+          OSName: 'Linux'
+          OSVmImage: 'ubuntu-16.04'
+          PythonVersion: 'pypy3'
+        MacOS_Python35:
```
Comment (danieljurek) [inline eng/pipelines/templates/jobs/archetype-sdk-nightly.yml]: These names should match the OS and Python versions

Probably should be 
`Windows_Python27` and `MacOS_Python35` respectively

Comment (scbedd) [reply eng/pipelines/templates/jobs/archetype-sdk-nightly.yml]: Yeah this is fixed in another commit I haven't pushed yet. Thanks for the catch.

Comment (scbedd) [reply eng/pipelines/templates/jobs/archetype-sdk-nightly.yml]: (Attempting to get rid of duplication, only to hit template insertion errors)

File: eng/pipelines/templates/jobs/archetype-sdk-nightly.yml [hunk 2]
```diff
@@ -88,38 +72,53 @@ jobs:
         ToxTestEnv: "whl,sdist"  
     - template: ../steps/publish-coverage.yml
 
-  # Run PyPy tests without coverage
-  - job: 'Test_PyPy'
-
+  - job: 'Test_No_Coverage'
```
[the remark this answers is not in the document]
Comment (mitchdenny) [reply eng/pipelines/templates/jobs/archetype-sdk-nightly.yml]: Yeah - I'd definitely go with just setting this as a matrix parameter.

[the remark this answers is not in the document]
Comment (scbedd) [reply eng/pipelines/templates/jobs/archetype-sdk-nightly.yml]: Doin it.

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 3]
```diff
@@ -45,26 +45,37 @@ jobs:
           OSName: 'Linux'
           OSVmImage: 'ubuntu-16.04'
           PythonVersion: '2.7'
+          CoverageArg: ''
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml]: Can this be setup as a global variable and then only override it in the case you want to set it? That would help reduce the redundancy.

**Pull request merged**